erpnext/hr/doctype/expense_claim/expense_claim.js

```diff
@@ -229,36 +234,78 @@ frappe.ui.form.on("Expense Claim", {
 	},
 
 	employee_name: function(frm) {
-		erpnext.expense_claim.set_title(frm);
+		erpnext.hr.expense_claim.set_title(frm);
 	},
 
 	task: function(frm) {
-		erpnext.expense_claim.set_title(frm);
+		erpnext.hr.expense_claim.set_title(frm);
 	}
 });
 
+$.extend(cur_frm.cscript, new erpnext.hr.ExpenseClaimController({frm: cur_frm}));
+
+erpnext.hr.expense_claim = {
+	set_title: function(frm) {
+		if (!frm.doc.task) {
+			frm.set_value("title", frm.doc.employee_name);
+		}
+		else {
+			frm.set_value("title", frm.doc.employee_name + " for "+ frm.doc.task);
+		}
+	},
+	clear_sanctioned: function(doc) {
+		var val = doc.expenses || [];
+		for(var i = 0; i<val.length; i++){
+			val[i].sanctioned_amount ='';
+		}
+
+		doc.total_sanctioned_amount = '';
+		refresh_many(['sanctioned_amount', 'total_sanctioned_amount']);
+	},
+	set_help: function(frm) {
+		frm.set_intro("");
+		if((frm.doc.__islocal && !in_list(frappe.user_roles, "HR User")) ||(frm.doc.__unsaved)) {
+
+			frm.set_intro(__("Fill the form and save it"));
+		} else {
+			if(frm.doc.docstatus==0 && frm.doc.approval_status=="Draft") {
+				if(frappe.session.user==frm.doc.exp_approver) {
+					frm.set_intro(__("You are the Expense Approver for this record. Please Update the 'Status' and Save"));
+				} else {
+					frm.set_intro(__("Expense Claim is pending approval. Only the Expense Approver can update status."));
+				}
+			}
+		}
+	}
+}
+
 frappe.ui.form.on("Expense Claim Detail", {
 	claim_amount: function(frm, cdt, cdn) {
 		var child = locals[cdt][cdn];
 		var doc = frm.doc;
 
-		if(!child.sanctioned_amount){
+		if((!child.sanctioned_amount) || (child.sanctioned_amount > child.claim_amount)){
 			frappe.model.set_value(cdt, cdn, 'sanctioned_amount', child.claim_amount);
 		}
 
-		cur_frm.cscript.calculate_total(doc,cdt,cdn);
+		frappe.model.set_value(cdt, cdn, 'sanctioned_tax', child.tax_amount * child.sanctioned_amount / child.claim_amount);
```

Comment:
Tax is included in both sanctioned amount and clam amount, so I think the formula is correct, no?